user_token.go

```diff
@@ -0,0 +1,136 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ UserTokens = (*userTokens)(nil)
+
+// UserTokens describes all the user token related methods that the
+// Terraform Cloud/Enterprise API supports.
+//
+// TFE API docs:
+// https://www.terraform.io/docs/enterprise/api/user-tokens.html
+type UserTokens interface {
+	// List all the tokens of the given user ID.
+	List(ctx context.Context, userID string) (*UserTokenList, error)
+
+	// Create a new user token
+	Create(ctx context.Context, userID string, options UserTokenCreateOptions) (*UserToken, error)
```

Comment:
With the understanding that I actually _like_ `Create()`...

The APIs for team and organization tokens use [`Generate()`](https://github.com/hashicorp/go-tfe/blob/19dc0b8eec47f6889541bd5b4ee6e4a81b48a908/team_token.go#L21). While I disagree with how those were named, consistency might be more important.

Reply:
I went with `Create()` because the API docs referred to the action as "Create": https://www.terraform.io/docs/cloud/api/user-tokens.html#create-a-user-token 

That said, the actual endpoint doesn't really mention it so I favor consistency too, and changed it to `Generate()` in 486d641